auth/oidc_test.go

```diff
@@ -378,3 +380,75 @@ func TestOIDCToHTTPError(t *testing.T) {
 	assert.Error(t, httpErr)
 	assert.Contains(t, httpErr.Error(), strconv.Itoa(http.StatusBadRequest))
 }
+
+func TestSetURLQueryParam(t *testing.T) {
+	var oidcAuthProviderGoogle = "google"
+	tests := []struct {
+		name             string
+		inputCallbackURL string
+		inputParamName   string
+		inputParamValue  string
+		wantCallbackURL  string
+		wantError        error
+	}{{
+		name:             "Add provider to callback URL",
+		inputCallbackURL: "http://localhost:4984/default/_oidc_callback",
+		inputParamName:   OIDCAuthProvider,
+		inputParamValue:  oidcAuthProviderGoogle,
+		wantCallbackURL:  "http://localhost:4984/default/_oidc_callback?provider=google",
```

Comment:
Can we add a test case for a provider name that would need URL encoding?

Something like `test&provider` ? It seems like there's nothing preventing provider names containing anything other than underscores, so we need to make sure they're being escaped properly in the callback URL.

Reply:
Added test case for a provider name that would need URL encoding.